### ISSUE TITLE
build fails due to DNS lookup error

### ISSUE THREAD

**Issue by mikevm** [Shippable/support#2252]
tl;dr: DNS lookups succeed several times, and then they fail, all for the same hostname, within a single rspec in a single build

Over the last several months I have frequently seen builds fail due to DNS lookup problems.  This is always happening during my `rspec` run (which is really the only thing that my build does).

`rspec` will run several tests where the DNS lookup succeeds, and then another test will have the DNS lookup fail, even though it is the same name that is being looked up.  The lookup usually fails when looking up the IP address of my AWS RDS instance.

> DataObjects::ConnectionError:
       could not translate host name "[redacted].us-west-1.rds.amazonaws.com" to address: Name or service not known

.

> DataObjects::SQLError:
       Unknown MySQL server host '[redacted].us-west-1.rds.amazonaws.com' (0)

If you look at these two builds: 56c4b098fb50fb0c00b5f876 56c4b6a7415c0d0c00ef5a05 you will see that they are both building the same commit from my repo, and they fail at different points in the build, both due to dns lookup failures that succeeded earlier in the build (since the same two databases are used in almost every one of my `rspec` examples).

I don't actually know how the DNS lookups happen - I'm not installing any package like `nslookup` or `bind` so I assume that the docker config provides this for me.

I have also seen in other builds (sorry no references right now, but I can hunt them down if it helps) that `bundle install` fails because ruby can't lookup `rubygems.org`.  Thus, the problem is not confined only to `rspec`, is not confined to my own ruby code, and is not confined to looking up AWS RDS hostnames.

Is there any reason to think that the DNS lookups are failing due to connectivity issues to the internet from my container?  Is there a way to encourage it to retry?  Is there something else that I have missed?

Thanks for any advice.

**Comment by mikevm** [comment]
I've now had 16 builds in a row fail due to DNS lookup failures in my builds, never in the same place twice.  Curious if any other folks experience similar issues.

**Comment by manishas** [comment]
Sorry for the delay in getting to this. We released an upgraded build system last week and that will resolve the issues you're seeing. Every build now runs in a CI container on a separate machine and we have also switched hosting providers, so connectivity should no longer be an issue.

I am closing this issue but let us know if we can help get you back on track.
